irods/test/user_group_test.py

```diff
@@ -418,6 +440,55 @@ def test_group_admin_can_create_users__issue_428(self):
             if rodsuser:
                 admin.users.remove(rodsuser.name)
 
+    # Test to ensure a rodsuser cannot lock themselves out by corrupting the catalog entry containing their
+    # encoded password.
+
+    # Need to record a separate issue for this.
+
+    def test_against_database_password_corruption_in_modify_password__issue_XYZ(self):
+        ses = helpers.make_session()
+        if ses.users.get( ses.username ).type != 'rodsadmin':
+            self.skipTest( 'Only a rodsadmin may run this test.')
+
+        user_alice = None
+        extra="."
+        NEWPASS='1234567890123456789012345'
+        OLDPASS='123456789012345678901234'
+
+        try:
+            user_alice = ses.users.create('alice', 'rodsuser')
+            ses.users.modify('alice', 'password', OLDPASS)
+
+            def alice_login(pw):
+                with iRODSSession(user='alice', password=pw, host=ses.host, port=ses.port, zone=ses.zone) as alice:
+                    home = helpers.home_collection( alice )
+                    alice.collections.get(home)
+
+            alice_login(OLDPASS)
+
+            # Alice makes an unsuccessful attempt at changing her own password.
+            with iRODSSession(user='alice', password=OLDPASS, host=ses.host, port=ses.port, zone=ses.zone) as alice:
+                me = alice.users.get(alice.username)
+                try:
+                    me.modify_password(OLDPASS+extra, NEWPASS)
+                except (ex.CAT_PASSWORD_ENCODING_ERROR, Bad_password_change_parameter):
```

Comment:
Does this mean we don't know which exception will be thrown?

Reply:
It actually means different exception will be thrown for (4.2.11 | 4.3.0) than in (4.2.12 | 4.3.1)

Reply:
If that is about compatibility across iRODS versions, we should add a comment stating that. Otherwise, we'll forget why this was needed in 6 months to a year.

Reply:
Will do

Reply:
This will now throw `Bad_password_change_parameter`, the reason being I've updated the client library to screen parameters that would cause either server error. I've placed an explanation in the commit text that various errors can be generated from the server if this is not done.

The fact that sometimes the server modifies the DB password entry destructively, without any warning (ie no server error happens at all) is a deeper problem that I don't know we'll be solving soon .  It would involve changing our native password encoding / decoding / obfuscation-over-network protocol profoundly.

Reply:
Seems good to me. The idea being that even though different results come back from the server across the aforementioned versions, this client will behave consistently (i.e. throwing `Bad_password_change_parameter`)?

Reply:
That's fine with me.

Your commit message needs a small tweak.
- "if an incorrect value**s**" -> "if an incorrect value"

Please mark this resolved once the commit message is updated.

Reply:
@alanking yes, tests as written have shown consistent behavior across iRODS server versions.